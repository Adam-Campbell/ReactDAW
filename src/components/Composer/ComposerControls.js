import React from 'react';
import PropTypes from 'prop-types';
import SelectInput from '../SelectInput';

const ComposerControls = props => (
    <div className="composer__controls-container">
        <SelectInput
            inputId="composer-cursor-select"
            label="Select tool:"
            value={props.cursorValue}
            handleChange={props.updateCursorValue}
            options={[
                { value: 'pointer', text: 'Pointer' },
                { value: 'pencil', text: 'Pencil' }
            ]} 
        />
        <SelectInput 
            inputId="composer-duration-select"
            label="Select section duration:"
            value={props.durationValue.toString()}
            handleChange={props.updateDurationValue}
            options={[
                { value: '1', text: '1' },
                { value: '2', text: '2' },
                { value: '3', text: '3' },
                { value: '4', text: '4' }
            ]} 
        />
    </div>
);

ComposerControls.propTypes = {
    cursorValue: PropTypes.string.isRequired,
    updateCursorValue: PropTypes.func.isRequired,
    durationValue: PropTypes.number.isRequired,
    updateDurationValue: PropTypes.func.isRequired
};

export default ComposerControls;