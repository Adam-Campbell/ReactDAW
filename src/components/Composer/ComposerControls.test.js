import React from 'react';
import { shallow } from 'enzyme';
import ComposerControls from './ComposerControls';

test('renders correctly', () => {
    const component = shallow(
        <ComposerControls 
            cursorValue="pointer"
            updateCursorValue={jest.fn()}
            durationValue={4}
            updateDurationValue={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});