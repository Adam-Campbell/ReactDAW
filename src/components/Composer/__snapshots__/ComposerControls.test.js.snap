// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="composer__controls-container"
>
  <SelectInput
    handleChange={[MockFunction]}
    inputId="composer-cursor-select"
    label="Select tool:"
    options={
      Array [
        Object {
          "text": "Pointer",
          "value": "pointer",
        },
        Object {
          "text": "Pencil",
          "value": "pencil",
        },
      ]
    }
    value="pointer"
  />
  <SelectInput
    handleChange={[MockFunction]}
    inputId="composer-duration-select"
    label="Select section duration:"
    options={
      Array [
        Object {
          "text": "1",
          "value": "1",
        },
        Object {
          "text": "2",
          "value": "2",
        },
        Object {
          "text": "3",
          "value": "3",
        },
        Object {
          "text": "4",
          "value": "4",
        },
      ]
    }
    value="4"
  />
</div>
`;
